pkg/cmd/provisioning/aws/create_identity_provider.go

```diff
@@ -319,12 +374,133 @@ func createOIDCBucket(client aws.Client, bucketName, name, region, targetDir str
 				},
 			})
 			if err != nil {
-				return errors.Wrapf(err, "failed to tag the bucket %s", bucketName)
+				return "", errors.Wrapf(err, "failed to tag the bucket %s", bucketName)
+			}
+
+			if createPrivateS3 {
+				createOriginAccessIdentityOutput, err := client.CreateCloudFrontOriginAccessIdentity(&cloudfront.CreateCloudFrontOriginAccessIdentityInput{
+					CloudFrontOriginAccessIdentityConfig: &cloudfront.OriginAccessIdentityConfig{
+						CallerReference: awssdk.String(name),
+						Comment:         awssdk.String(fmt.Sprintf("%s/%s", ccoctlAWSResourceTagKeyPrefix, name)),
+					},
+				})
+				if err != nil {
+					return "", errors.Wrapf(err, "failed to create CloudFront origin access identity")
+				}
+				originAccessIdentityID := *createOriginAccessIdentityOutput.CloudFrontOriginAccessIdentity.Id
+				log.Printf("CloudFront origin access identity created with ID %s, waiting for 30 seconds for it to become active", originAccessIdentityID)
```

Comment:
nit:
```suggestion
				log.Printf("CloudFront origin access identity created with ID %s, waiting for %s for it to become active", originAccessIdentityID, cloudFrontOriginAccessIdentityActivationGracePeriod)
```
...in case the value of that const changes.